### ISSUE TITLE
2024 Curriculum Overhaul (WIP)

### ISSUE THREAD

**Issue by adrianmcli** [privacy-scaling-explorations/contribution-program#24]
# Summary

PSE EcoDev had a mini off-site session in February 2024 to work on overhauling the curriculum structure for 2024. The main changes are:

1. Move easy sections to a "pre-requisite" module before program start.
2. The "Practical Track" and "Study Track" will happen in parallel to encourage coding from the start.
3. Certain concepts are "black-boxed" and links are placed in an optional section.
4. Mini-hackathon will be placed in the middle of the program.

# New Structure

### Course Info
1. About This Course
2. Open-Source Culture, Git, and Github
4. How to ask questions, where to find help
5. How to submit your assignments (PR workflow)
6. Course FAQ

### Knowledge
1. A Primer for ZKPs
2. Thought Experiments
3. Use-Cases and Applications
4. Supplemental materials on ZKPs [optional]
5. Number Theory

### Study Track

1. Symmetric vs Asymmetric Encryption (AES, RSA)
2. Hash Functions, Merkle Trees
3. Digital Signatures (Schnorr)
4. DLP-based Public-Key Cryptography (DLP, DH, Elgamal)
5. Cryptographic Commitments
6. Intro to Elliptic Curve Cryptography
7. BLS Signatures (keep this simple and black-boxed)
8. Pairing-Based Cryptography (keep this simple and black-boxed)

## 2️⃣ Week 2 - More Crypto and zkSNARKs
1. KZG Polynomial Commitments (keep this simple and black-boxed)
2. Trusted Setups (keep this simple and black-boxed; reference Inevitable Ethereum)
3. What is a zkSNARK?
4. From Computation to ZKPs (use only Vitalik's blog post; arithmetic circuits, R1CS, QAP)
5. Groth16 (keep this simple and black-boxed)

## Week 5 - 

# Reference Pictures from Off-Site

## Practical Track
![2024-02-07 15 24 33](https://github.com/privacy-scaling-explorations/contribution-program/assets/943555/bf9b4aa2-48b4-47e4-80ef-d28517687200)

## Study Track
![2024-02-07 15 24 47](https://github.com/privacy-scaling-explorations/contribution-program/assets/943555/8d6a9a0f-852c-48b6-87e4-5796b70e0e67)

**Comment by NOOMA-42** [comment]
Let's make it a ticket table with checkbox

**Comment by aguzmant103** [comment]
@adrianmcli will this be merged into the repo? It'd be nice to tag as releases the versions and see 2023 vs 2024 vs 2025 curriculum versions :)

**Comment by NOOMA-42** [comment]
We reorganize the folder and make 2023 2024 separate
> @adrianmcli will this be merged into the repo? It'd be nice to tag as releases the versions and see 2023 vs 2024 vs 2025 curriculum versions :)
